pkg/netscan/discover.go

```diff
@@ -0,0 +1,271 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package netscan
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"github.com/pkg/errors"
+	"math"
+	"net"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+)
+
+const (
+	NetworkUDP = "udp"
+	NetworkTCP = "tcp"
+)
+
+// AutoDiscover probes all addresses in the configured network to attempt to discover any possible
+// devices for a specific protocol
+func AutoDiscover(ctx context.Context, proto ProtocolSpecificDiscovery, params Params) []DiscoveredDevice {
+	params.Logger.Debugf("AutoDiscover called with the following parameters: %+v", params)
+	if len(params.Subnets) == 0 {
+		params.Logger.Warnf("Discover was called, but no subnet information has been configured!")
```

Comment:
```suggestion
		params.Logger.Warn("Discover was called, but no subnet information has been configured!")
```

Reply:
That function doesn't exist on the interface. But if I switch to EdgeX logger I will remove the `f`.